CHANGELOG.md

```diff
@@ -1,5 +1,20 @@
 # libhoney Changelog
 
+## 1.23.0 2024-06-05
+
+### Enhancements
+
+- feat: URL encode dataset (#242) | @MikeGoldsmith
+
+### Maintenance
+
+- maint: add labels to release.yml for auto-generated grouping (#241) | @JamieDanielson
+- maint: Update minimum go version to 1.19 | @MikeGoldsmith
```

Comment:
Can we mark this as a breaking change similar to how we did this last time? https://github.com/honeycombio/libhoney-go/blob/efa6b7a4d2be7665f1678d84238fcd3a763037f3/CHANGELOG.md?plain=1#L30-L34